x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openshiftai/OpenShiftAiService.java

```diff
@@ -0,0 +1,427 @@
+            chunkingSettingsMap = ChunkingSettingsBuilder.fromMap(
+                removeFromMapOrDefaultEmpty(config, ModelConfigurations.CHUNKING_SETTINGS)
+            );
+        }
+        Map<String, Object> taskSettingsMap = removeFromMapOrDefaultEmpty(config, ModelConfigurations.TASK_SETTINGS);
+
+        return createModelFromPersistent(inferenceEntityId, taskType, serviceSettingsMap, null, taskSettingsMap, chunkingSettingsMap);
+    }
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return OpenShiftAiUtils.ML_INFERENCE_OPENSHIFT_AI_ADDED;
+    }
+
+    @Override
+    public Set<TaskType> supportedStreamingTasks() {
+        return EnumSet.of(TaskType.COMPLETION, TaskType.CHAT_COMPLETION);
+    }
+
+    private static OpenShiftAiModel createModel(
+        String inferenceEntityId,
+        TaskType taskType,
+        Map<String, Object> serviceSettings,
+        @Nullable Map<String, Object> secretSettings,
+        Map<String, Object> taskSettings,
+        ChunkingSettings chunkingSettings,
+        ConfigurationParseContext context
+    ) {
+        return switch (taskType) {
+            case CHAT_COMPLETION, COMPLETION -> new OpenShiftAiChatCompletionModel(
+                inferenceEntityId,
+                taskType,
+                NAME,
+                serviceSettings,
+                secretSettings,
+                context
+            );
+            case TEXT_EMBEDDING -> new OpenShiftAiEmbeddingsModel(
+                inferenceEntityId,
+                taskType,
+                NAME,
+                serviceSettings,
+                chunkingSettings,
+                secretSettings,
+                context
+            );
+            case RERANK -> new OpenShiftAiRerankModel(
+                inferenceEntityId,
+                taskType,
+                NAME,
+                serviceSettings,
+                taskSettings,
+                secretSettings,
+                context
+            );
+            default -> throw createInvalidTaskTypeException(inferenceEntityId, NAME, taskType, context);
+        };
+    }
+
+    private OpenShiftAiModel createModelFromPersistent(
+        String inferenceEntityId,
+        TaskType taskType,
+        Map<String, Object> serviceSettings,
+        Map<String, Object> secretSettings,
+        Map<String, Object> taskSettings,
+        ChunkingSettings chunkingSettings
+    ) {
+        return createModel(
+            inferenceEntityId,
+            taskType,
+            serviceSettings,
+            secretSettings,
+            taskSettings,
+            chunkingSettings,
+            ConfigurationParseContext.PERSISTENT
+        );
+    }
+
+    @Override
+    public int rerankerWindowSize(String modelId) {
+        // OpenShift AI uses Cohere and JinaAI rerank protocols for reranking
+        // JinaAI rerank model has 8000 tokens limit length https://jina.ai/models/jina-reranker-v2-base-multilingual
```

Comment:
The Jina V2 reranker is deprecated by the [newer V3 reranker](https://jina.ai/models/jina-reranker-v3), which has an input token length of 131K. This doesn't change the value we return from this method, since Cohere still has a smaller limit, but it might be good to link to the latest docs here anyway.